src/Dictionaries/HashedDictionary.cpp

```diff
@@ -614,9 +797,10 @@ void HashedDictionary<dictionary_key_type, sparse>::getItemsImpl(
     {
         auto key = keys_extractor.extractCurrentKey();
 
-        const auto it = attribute_container.find(key);
+        const auto & container = attribute_containers[getShard(key)];
```

Comment:
In practice this should make `dictHas`, `dictGet`, and other functions that work with hierarchy slower at least multiple times when there is sharding. In worst case it can work `shards` times slower.

Reply:
I don't see why it should be slower, and I've just tested it - can't confirm, I will submit perf test for this.

But frankly speaking I wouldn't expect shards usage for hierarchical dictionaries, since it is only makes sense for big enough dictionaries, while I doubt that hierarchical used in this way.